ctapipe/core/tests/test_tool.py

```diff
@@ -76,5 +81,33 @@ class MyTool(Tool):
     tool.userparam = -1.0
     conf2 = tool.get_current_config()
 
-    assert conf1['MyTool']['userparam'] == 5.0
-    assert conf2['MyTool']['userparam'] == -1.0
+    assert conf1["MyTool"]["userparam"] == 5.0
+    assert conf2["MyTool"]["userparam"] == -1.0
+
+
+def test_tool_command_line_precedence():
+    """
+    ensure command-line has higher priority than config file
+    """
+
+    class SubComponent(Component):
+        component_param = Float(10.0, help="some parameter").tag(config=True)
+
+    class MyTool(Tool):
+        description = "test"
+        userparam = Float(5.0, help="parameter").tag(config=True)
+
+        classes = List([SubComponent,])
+        aliases = Dict({"component_param": "SubComponent.component_param"})
+
+        def setup(self):
+            self.sub = self.add_component(SubComponent(parent=self))
+
+    config = Config(
+        {"MyTool": {"userparam": 12.0}, "SubComponent": {"component_param": 15.0}}
+    )
+
+    tool = MyTool(config=config)  # sets component_param to 15.0
+
+    tool.run(argv=["--component_param", "20.0"])
+    assert tool.sub.component_param == 20.0
```

Comment:
Maybe also test that a config option that was not given on the cli but as `Config` has the correct value and not the default?

Reply:
that's covered by other tests already, but could add it again in case somehow the command-line breaks the config functionality. .

Reply:
I thought that maybe the update might reset stuff to the defaults if not given in the cli. but if that's not the case or tested elsewhere, great

Reply:
I'll add it just in case. traitlets.config is so poorly documented, that who knows.